File: 2492-Minimum Score of a Path Between Two Cities.py
class Solution:
    def minScore(self, n: int, roads: List[List[int]]) -> int:
        hmap = defaultdict(list)
        for u, v, w in roads:
            hmap[u].append((v, w))
            hmap[v].append((u, w))
            
        used = set([1])
        queue = deque([1])
        res = float('inf')
        
        while queue:
            curr = queue.popleft()
            for nxt, w in hmap[curr]:
                if nxt not in used:
                    used.add(nxt)
                    queue.append(nxt)
                res = min(res, w)
        return res

File: C++/2492-Minimum Score of a Path Between Two Cities.cpp
class Solution {
public:
    int res = INT_MAX;
    void dfs(int curr, vector<bool>& visited, vector<vector<pair<int, int>>>& graph) {
        
        for (auto& [nxt, w]:graph[curr]) {
            res = min(res, w);
            if (visited[nxt]) continue;
            visited[nxt] = true;
            dfs(nxt, visited, graph);
        }
    }

    int minScore(int n, vector<vector<int>>& roads) {
        vector<vector<pair<int, int>>> graph(n+1);
        vector<bool> visited(n+1, false);
        for (auto &r:roads) {
            graph[r[0]].push_back({r[1], r[2]});
            graph[r[1]].push_back({r[0], r[2]});
        }

        visited[1] = true;
        dfs(1, visited, graph);
        
        return res;
    }
};